utilities/transactions/pessimistic_transaction_db.cc

```diff
@@ -600,7 +601,39 @@ void WritePreparedTxnDB::AddCommitted(uint64_t prepare_seq,
       {
         WriteLock wl(&snapshots_mutex_);
         InstrumentedMutex(db_impl_->mutex());
-        snapshots_ = db_impl_->snapshots().GetAll();
+        // We only care about snapshots lower then max
+        auto all_snapshots =
+            db_impl_->snapshots().GetAll(nullptr, max_evicted_seq);
+        auto curr_seq = db_impl_->GetLatestSequenceNumber();
+        // This is to avoid updating the snapshots_ if it already updated with a
+        // more recent vesion by a concrrent thread
+        if (curr_seq > snapshots_version_) {
+          snapshots_version_ = curr_seq;
+          // Both new and old lists are sorted and the new list is subset of the
+          // previous list plus some new items. Thus if a snapshot repeats in
+          // both new and old lists, it will appear upper in the new list. So if
```

Comment:
Why it will appear upper in the new list? Snapshots can be released out of order, say you have
old list: 1,2,3
and then 1 is released then you end up with 
new list: 2,3